src/tools/tests/allocator-recipe-resolver-test.ts

```diff
@@ -1149,6 +1157,167 @@ recipe ReadingRecipeAD
       ${recipe}
     `, 'Thing {a: (Text, Number), c: List<(Text, Number)>, b: [(Text, Number)], d: (Text, Number)}');
   });
+
+  it.only('restricts writer fields with phantom readers', async () => {
+    const verifyPhantomReader = async (
```

Comment:
suggestion - maybe factor out the `verifyPhantomReader` method outside the test case and break the test case to smaller more focused ones?

Reply:
Thanks, Maria! I have split the tests into smaller focussed ones.